motorway/mixins.py

```diff
@@ -44,12 +48,23 @@ def send_message(self, message, process_id, time_consumed=None, sender=None, con
         except GroupingValueMissing:
             raise GroupingValueMissing("Message '%s' provided an invalid grouping_value: '%s'" % (message.content, message.grouping_value))
         for destination in socket_addresses:
-            message.send(
-                self.send_socks[destination],
-                process_id
-            )
+            # Keep trying to send the message until successful
+            message_retries = 0
+            while True:
+                try:
+                    message.send(
+                        self.send_socks[destination],
+                        process_id
+                    )
+                    break
+                # Queue was full or not available
+                except zmq.Again as e:
+                    message_retries +=1
+                    logger.warning("Failed to send message from %s (retry # %s)", process_id, message_retries)
+                    time.sleep(5)
```

Comment:
5 seconds? That's a long sleep

Reply:
Is it fine to retry forever or should we implement some limit?

Reply:
It's fine to retry forever, it will essentially be the same behavior as we have now, where the call just blocks until a message can be delivered. If we don't retry forever we will start breaking the pipeline when we have an exception

Reply:
Based on tests, it takes a while before something can be added to the queue (more than one item has to be taken off).
So this is to avoid flooding logs.